code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1312,5 +1319,148 @@
+	..()
+	. = 1
+
+/datum/reagent/medicine/trophazole/overdose_process(mob/living/M)
+	M.adjustBruteLoss(3*REM, 0)
+	..()
+	. = 1
+
+/datum/reagent/medicine/trophazole/reaction_mob(mob/living/carbon/M, method=INGEST, reac_volume)
+	if(method != INGEST)
+		return
+
+	M.reagents.remove_reagent(/datum/reagent/medicine/trophazole, reac_volume * 0.05)
+	M.reagents.add_reagent(/datum/reagent/medicine/metafactor, reac_volume * 0.25)
+
+	..()
+
+/datum/reagent/medicine/metafactor
+	name = "Mitogen Metabolism Factor"
+	description = "This enzyme catalyzes the conversion of nutricious food into healing peptides."
+	metabolization_rate = 0.0625  * REAGENTS_METABOLISM //slow metabolism rate so the patient can self heal with food even after the troph has metabolized away for amazing reagent efficency.
+	reagent_state = SOLID
+	color = "#DC605D"
+	overdose_threshold = 10
+
+/datum/reagent/medicine/metafactor/overdose_start(mob/living/carbon/M)
+	metabolization_rate = 2  * REAGENTS_METABOLISM
+
+/datum/reagent/medicine/metafactor/overdose_process(mob/living/carbon/M)
+	if(prob(25))
+		M.vomit()
+	..()
+
+/datum/reagent/medicine/rhigoxane
+	name = "Rhigoxane"
+	description = "A second generation burn treatment agent exibiting a cooling effect that is especially pronounced when deployed as a spray. It's high halogen content helps extiguish fires."
+	reagent_state = LIQUID
+	color = "#B6D2F2"
+	overdose_threshold = 25
+	reagent_weight = 0.6
+
+/datum/reagent/medicine/rhigoxane/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(-2*REM, 0.)
+	M.adjust_bodytemperature(-20 * TEMPERATURE_DAMAGE_COEFFICIENT, BODYTEMP_NORMAL)
+	..()
+	. = 1
+
+/datum/reagent/medicine/rhigoxane/reaction_mob(mob/living/carbon/M, method=VAPOR, reac_volume)
+	if(method != VAPOR)
+		return
+
+	M.adjust_bodytemperature(-reac_volume * TEMPERATURE_DAMAGE_COEFFICIENT * 20, 200)
+	M.adjust_fire_stacks(-reac_volume / 2)
+	if(reac_volume >= metabolization_rate)
+		M.ExtinguishMob()
+
+	..()
+
+/datum/reagent/medicine/rhigoxane/overdose_process(mob/living/carbon/M)
+	M.adjustFireLoss(3*REM, 0.)
+	M.adjust_bodytemperature(-35 * TEMPERATURE_DAMAGE_COEFFICIENT, 50)
+	..()
+
+/datum/reagent/medicine/thializid
+	name = "Thializid"
+	description = "A potent antidote for intravenous use with a narrow therapeutic index, it is considered an active prodrug of oxalizid."
+	reagent_state = LIQUID
+	color = "#8CDF24" // heavy saturation to make the color blend better
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	overdose_threshold = 6
+	var/conversion_amount
+
+/datum/reagent/medicine/thializid/reaction_mob(mob/living/carbon/M, method=INJECT, reac_volume)
+	if(method != INJECT)
+		return
+	if(reac_volume >= 0.6) //prevents cheesing with ultralow doses.
+		M.adjustToxLoss(-1.5 * min(2, reac_volume) * REM, 0)	  //This is to promote iv pole use for that chemotherapy feel.
+	if(!locate(/obj/item/organ/liver) in M.internal_organs)
+		return
+	var/obj/item/organ/liver/L = M.internal_organs_slot[ORGAN_SLOT_LIVER]
```

Comment:
It would probably be better to remove the above if statement on L#1408, and replace it with a null check against L.  That way you only iterate through organs once, and incase someone somehow shoves a liver in the wrong slot, it won't break.